cyto_dl/api/cyto_dl_model/cyto_dl_base_model.py

```diff
@@ -0,0 +1,135 @@
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, List, Optional, Union
+
+import pyrootutils
+from hydra import compose, initialize_config_dir
+from hydra.core.global_hydra import GlobalHydra
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from cyto_dl.api.data import *
+from cyto_dl.eval import evaluate as evaluate_model
+from cyto_dl.train import train as train_model
+
+
+class CytoDLBaseModel(ABC):
+    """A CytoDLBaseModel is used to configure, train, and run predictions on a cyto-dl model."""
+
+    def __init__(self, config_filepath: Optional[Path] = None):
+        """
+        :param config_filepath: path to a .yaml config file that will be used as the basis
+        for this CytoDLBaseModel (must be generated by the CytoDLBaseModel subclass that wants
+        to use it). If None, a default config will be used instead.
+        """
+        self._cfg: DictConfig = (
+            OmegaConf.load(config_filepath) if config_filepath else self._generate_default_config()
+        )
+
+    def _generate_default_config(self) -> DictConfig:
+        cfg_dir: Path = (
+            pyrootutils.find_root(search_from=__file__, indicator=("pyproject.toml", "README.md"))
+            / "configs"
+        )
+        GlobalHydra.instance().clear()
+        with initialize_config_dir(version_base="1.2", config_dir=str(cfg_dir)):
+            # the overrides for 'paths.' are necessary to make later overrides work (for some reason I don't fully understand)
+            # I'd prefer to do this in experiment configs instead of the overrides here
```

Comment:
@benjijamorris lmk if you think it's fine to make these overrides in `experiment/im2im/segmentation_plugin.yaml`. I'm imagining something like
```
paths:
    output_dir: PLACEHOLDER
    work_dir: PLACEHOLDER
```

Reply:
sounds good, I think this will be fixed by moving all overrides to the top level config